src/lidar_tools/dsm_functions.py

```diff
@@ -50,51 +85,122 @@ def return_readers(input_aoi,
             # gdf = gpd.read_file('../data/shapefiles/resources.geojson').set_crs(4326)
-
+            if return_specific_3dep_survey is not None:
+                return_all_intersecting_surveys = True
             for _, row in gdf.iterrows():
                 if row.geometry.intersects(aoi_4326):
                     usgs_dataset_name = row['name']
-                    break
-
-            print("Dataset being used: ", usgs_dataset_name)
-            url = f"https://s3-us-west-2.amazonaws.com/usgs-lidar-public/{usgs_dataset_name}/ept.json"
-            reader = {
-            "type": "readers.ept",
-            "filename": url,
-            "resolution": pointcloud_resolution,
-            "polygon": str(aoi_3857.wkt),
-            }
-
-            # SRS associated with the 3DEP dataset
-            response = requests.get(url)
-            data = response.json()
-            srs_wkt = data['srs']['wkt']
-
-            pointcloud_input_crs.append(CRS.from_wkt(srs_wkt))
-            readers.append(reader)
+                    if return_specific_3dep_survey is not None:
+                        if usgs_dataset_name == return_specific_3dep_survey:
+                            add_survey = True
+                        else:
+                            add_survey = False
+                    else:
+                        add_survey = True
+                    if add_survey:
+                        print("Dataset being used: ", usgs_dataset_name)
+                        url = f"https://s3-us-west-2.amazonaws.com/usgs-lidar-public/{usgs_dataset_name}/ept.json"
+                        reader = {
+                        "type": "readers.ept",
+                        "filename": url,
+                        "resolution": pointcloud_resolution,
+                        "polygon": str(aoi_3857.wkt),
+                        }
+
+                        # SRS associated with the 3DEP dataset
+                        response = requests.get(url)
+                        data = response.json()
+                        srs_wkt = data['srs']['wkt']
+
+                        pointcloud_input_crs.append(CRS.from_wkt(srs_wkt))
+                        readers.append(reader)
+                        extents.append(aoi_3857.bounds)
+                        original_extents.append(src_bounds_transformed_3857)
+                    if not return_all_intersecting_surveys:
+                        break
+
+            
+
+    return readers, pointcloud_input_crs, extents, original_extents
+
+
+def return_reader_inclusive(input_aoi,
+                   src_crs,
+                   pointcloud_resolution=1):
+    """
+     This method takes an input aoi and finds overlapping 3DEP EPT data from https://s3-us-west-2.amazonaws.com/usgs-lidar-public/{usgs_dataset_name}/ept.json
+    It then returns a series of readers corresponding to non-overlapping areas for PDAL processing pipelines
 
+    Parameters
+    ----------
+    input_aoi : shapely.geometry.Polygon
```

Comment:
Not critical, but just noting you can move these type hints into the function definition, which has the advantage of given more feedback in code editors:
```
def return_reader_inclusive(input_aoi: shapely.geometry.Polygon,
                   src_crs: pyproj.CRS,
                   pointcloud_resolution: int=1):
```

Reply:
All functions have been now updated to include typehints in the function definition via: [1f9ca8c](https://github.com/uw-cryo/lidar_tools/commit/1f9ca8ca25240aeb4edecbd67822e163ade10700)